docs/design/node-group.md

```diff
@@ -0,0 +1,69 @@
+
+case1: different departments (nlp and recommend) use different nodes (nodegroup1 and nodegroup2), tts uses nodegroup1 first, and nodegroup2 can also be used when resources are insufficient, but recommend can only use nodegroup2.
+
+![](images/node-group-case-1.png)
+
+case2: recommend can use private cloud nodes or public cloud nodes, but tts can only use private cloud nodes because its task takes a long time, which is not easy to the release of public cloud nodes (for saving money)
+
+![](images/node-group-case-2.png)
+
+## Solution
+
+1. First, we need mark out some nodes(by add specific label for nodes) which are the same group.
+2. Second, we need to express the relationship between queue and nodegroup, such as affinity and antiaffinity
+   ```yaml
+   apiVersion: scheduling.volcano.sh/v1beta1
+   kind: Queue
+   metadata:
+     name: default
+     spec:
+       reclaimable: true
+       weight: 1
+       affinity:
+         nodeGroupAffinity:
+           requiredDuringSchedulingIgnoredDuringExecution:
+           - groupname1
+           - gropuname2
+           preferredDuringSchedulingIgnoredDuringExecution
+           - groupname1
+         nodeGroupAntiAffinity:
+           requiredDuringSchedulingIgnoredDuringExecution:
+           - groupname3
+           - gropuname4
+           preferredDuringSchedulingIgnoredDuringExecution
+           - groupname3
+         
+   ```
+
+affinity configure:
+1. affinity.nodeGroupAffinity.requiredDuringSchedulingIgnoredDuringExecution, hard constraints, such as `nlp = nodegroup1,nodegroup2`, it means that task in queue=nlp can ony run on the nodes in nodegroup1 or nodegroup2.
+2. affinity.nodeGroupAffinity.preferredDuringSchedulingIgnoredDuringExecution, soft constraints, such as `nlp = nodegroup1`, it means that task in queue=nlp runs on nodegroup1 first, but if the resources of nodegroup1 is insufficient, it can also run on other nodegroups.
+3. affinity.nodeGroupAntiAffinity.requiredDuringSchedulingIgnoredDuringExecution, hard constraints, such as `nlp = nodegroup1`, it means that task in queue=nlp can run on any nodegroups but nodegroup1.
+4. affinity.nodeGroupAntiAffinity.preferredDuringSchedulingIgnoredDuringExecution, soft constraints, such as `nlp = nodegroup1`, it means that task in queue=nlp runs on any other nodegroups, but if the resources of other nodegroup1 is insufficient, it can also run on nodegroup1.
+
+example
+
+```yaml
+actions: "reclaim, allocate, backfill, preempt"
+tiers:
+- plugins:
+  - name: priority
+  - name: gang
+  - name: conformance
+- plugins:
+  - name: drf
+  - name: predicates
+  - name: proportion
+  - name: nodegroup
+    enablePredicate: true
+    enableNodeOrder: true
+```
+
+risk: The resources of the queue can not be too different from the resources of the node-group, otherwise it is easy that task can be scheduled to run from the perspective of the queue, but cannot find a suitable node.
+
```

Comment:
Please clariy the following thing in the design doc
1.  Queue API change:
2.  Command line: 
     both kubectl and vcctl support display nodegroup info in queue querying via CLI ?